pkg/app/piped/cloudprovider/kubernetes/diff.go

```diff
@@ -75,6 +86,44 @@ func DiffList(olds, news []Manifest, opts ...diff.Option) (*DiffListResult, erro
 	return cr, nil
 }
 
+func normalizeNewSecret(old, new *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	var o, n v1.Secret
+	runtime.DefaultUnstructuredConverter.FromUnstructured(old.Object, &o)
+	runtime.DefaultUnstructuredConverter.FromUnstructured(new.Object, &n)
+
+	// Move as much as possible fields from `n.Data` to `n.StringData` to make `n` close to `o` to minimize the diff.
+	for k, v := range n.Data {
+		// Skip if the field also exists in StringData.
+		if _, ok := n.StringData[k]; ok {
+			continue
+		}
+
+		if _, ok := o.StringData[k]; !ok {
+			continue
+		}
+
+		if n.StringData == nil {
+			n.StringData = make(map[string]string)
+		}
+
+		// If the field is existing in `o.StringData`, we should move that field from `n.Data` to `n.StringData`
+		n.StringData[k] = string(v)
+		delete(n.Data, k)
+
+		// In case the `o.Data` is overridden, we also should restore `n.Data` from `o.Data`
+		if v, ok := o.Data[k]; ok {
+			n.Data[k] = v
+		}
```

Comment:
I don't think we need this. We should not copy any data from `o` to `n`.

Reply:
I fixed this. PTAL